dss-collection-nft/contracts/DSSCollection.cdc

```diff
@@ -0,0 +1,695 @@
+    )
+    pub event SlotCreated(
+        id: UInt64,
+        collectionGroupID: UInt64,
+        logicalOperator: String,
+        typeName: String
+    )
+    pub event ItemAddedToSlot(
+        id: UInt64,
+        slotID: UInt64,
+        collectionGroupID: UInt64
+    )
+    pub event CollectionNFTMinted(
+        id: UInt64,
+        collectionGroupID: UInt64,
+        serialNumber: UInt64,
+        completedBy: String,
+        completionDate: UFix64,
+        level: UInt8
+    )
+    pub event CollectionNFTBurned(id: UInt64)
+
+    // Named Paths
+    //
+    pub let CollectionStoragePath:  StoragePath
+    pub let CollectionPublicPath:   PublicPath
+    pub let AdminStoragePath:       StoragePath
+    pub let MinterPrivatePath:      PrivatePath
+
+    // Entity Counts
+    //
+    pub var totalSupply:    UInt64
+    pub var nextItemID:     UInt64
+
+    // Lists in contract
+    //
+    access(self) let collectionGroupByID: @{UInt64: CollectionGroup}
+    access(self) let slotByID: @{UInt64: Slot}
+    access(self) let itemByID: {UInt64: Item}
+
+    // A public struct to access Item data
+    //
+    pub struct Item {
+        pub let id: UInt64 // unique uuid of resource
+        pub let itemID: UInt64 // the id of the edition, tier, play
+        pub let points: UInt64 // points for item
+        pub let itemType: String // (edition.id, edition.tier, play.id)
+
+        init (
+            id: UInt64,
+            itemID: UInt64,
+            points: UInt64,
+            itemType: String
+        ) {
+            self.id = id
+            self.itemID = itemID
+            self.points = points
+            self.itemType = itemType
+
+            DSSCollection.nextItemID = self.id + 1 as UInt64
+
+            emit ItemCreated(
+                id: self.id,
+                itemID: self.itemID,
+                points: self.points,
+                itemType: self.itemType
+            )
+        }
+    }
+
+    // A public struct to access Slot data
+    //
+    pub struct SlotData {
+        pub let id: UInt64
+        pub let collectionGroupID: UInt64
+        pub let logicalOperator: String // (AND / OR)
+        pub let typeName: String // (A.contractAddress.NFT...)
+        pub var items: [Item]
+
+        init (id: UInt64) {
+            if let slot = &DSSCollection.slotByID[id] as &DSSCollection.Slot? {
+                self.id = slot.id
+                self.collectionGroupID = slot.collectionGroupID
+                self.logicalOperator = slot.logicalOperator
+                self.typeName = slot.typeName
+                self.items = slot.items
+            } else {
+                panic("slot does not exist")
+            }
+        }
+    }
+
+    // A top-level Slot with a unique ID
+    //
+    pub resource Slot {
+        pub let id: UInt64
+        pub let collectionGroupID: UInt64
+        pub let logicalOperator: String // (AND / OR)
+        pub let typeName: String // (A.contractAddress.NFT...)
```

Comment:
Again use `typeName` as `Type` variable.

Reply:
We do support parsing that.  Is it possible that one day this will be off-chain info though, like a badge or something?  If so, when we support off-chain info we would support it being a string here.  That would be odd to have on a contract though, so maybe type is best...